calculator.go

```diff
@@ -16,3 +18,12 @@ func Subtract(a, b float64) float64 {
 func Multiply(a, b float64) float64 {
 	return a * b
 }
+
+// Divide takes two numbers and returns the result by dividing the first by the second
+func Divide(a, b float64) (float64, error) {
+	if b != 0 {
```

Comment:
Excellent! Conventionally, we might write this the other way round, so that the 'happy path' is left-aligned.

```go 
if b == 0 {
	return 0, errors.New("division by zero not allowed")
}
```